carta/session.py

```diff
@@ -358,39 +359,62 @@ def set_starting_directory(self, path):
+        append : {2}
+            Whether the image should be appended. Default is ``False``.
         """
-        return Image.new(self, path, hdu, False, complex, expression)
+        path = self.resolve_file_path(path)
+        directory, file_name = posixpath.split(path)
+        return Image.new(self, directory, file_name, hdu, append, False)
 
-    @validate(String(), String(r"\d*"), Boolean(), Constant(ArithmeticExpression))
-    def append_image(self, path, hdu="", complex=False, expression=ArithmeticExpression.AMPLITUDE):
-        """Append a new image, keeping any existing images.
+    @validate(String(), String(r"\d*"), Constant(ComplexExpression), Boolean())
+    def open_complex_image(self, path, expression=ComplexExpression.AMPLITUDE, hdu="", append=False):
+        """Open or append a new complex-valued image.
 
         Parameters
         ----------
         path : {0}
-            The path to the image file, either relative to the session's current directory or an absolute path relative to the CARTA backend's root directory.
-        hdu : {1}
+            The path to the complex-valued image file, either relative to the session's current directory or an absolute path relative to the CARTA backend's root directory.
+        expression : {1}
+            The complex expression for opening a complex-valued image.
+        hdu : {2}
+            The HDU to select inside the file.
+        append : {3}
+            Whether the image should be appended. Default is ``False``.
+        """
+        path = self.resolve_file_path(path)
+        directory, file_name = posixpath.split(path)
+        file_name = f'{expression}("{file_name}")'
+        return Image.new(self, directory, file_name, hdu, append, True)
+
+    @validate(String(), String(), String(r"\d*"), Boolean())
+    def open_LEL_image(self, expression, directory, hdu="", append=False):
+        """Open or append a new image via the Lattice Expression Language (LEL) interface.
+
+        Parameters
+        ----------
+        expression : {0}
+            The LEL arithmetic expression.
+        directory : {1}
+            The directory to the image file(s), either relative to the session's current directory or an absolute path relative to the CARTA backend's root directory.
+        hdu : {2}
             The HDU to select inside the file.
-        complex : {2}
-            Whether the image is complex. Set to ``False`` by default.
-        expression : {3}
-            Arithmetic expression to use if appending a complex-valued image. The default is :obj:`carta.constants.ArithmeticExpression.AMPLITUDE`.
+        append : {3}
+            Whether the image should be appended. Default is ``False``.
         """
-        return Image.new(self, path, hdu, True, complex, expression)
+        file_name = expression
+        if "/" in expression:
+            directory = self.pwd()
```

Comment:
While it may be a good idea to default to the wrapper session's `pwd` rather than the GUI browser's current directory, what is the logic for this conditional? What are you trying to check for by checking if the expression contains `/`? It could be anywhere in the string, and doesn't mean that any or all of the paths are absolute. And I don't think that we should be overriding the directory that the user provides. It makes more sense for this to be the default.

If we do use `pwd` here instead of the current GUI browser directory, we should do the same in the stokes hypercube function.

Reply:
The reason I want to check if the `expression` contains `/` is that there are two types of `LEL` usage. 

The first one is that user provides path (both directories and files) in `expression`, like:
```
expression = '"/media/icchen/disk/fits/Orion_source_I_Stokes_I.fits"+"/media/icchen/disk/fits/Orion_source_I_Stokes_Q.fits"'
``` 
In this case, the `directory` argument doesn't really matter and we can adopt either `pwd` or `Macro("fileBrowserStore", "startingDirectory")` to make the function work.

The other one is that user only provide `file_name` in `expression`, like:
```
expression = '"Orion_source_I_Stokes_I.fits"+"Orion_source_I_Stokes_Q.fits"'
```
In this case, the `directory` argument **_does matter_** and we do need this `directory` argument to locate the files to open.

Therefore, I was trying to make different conditions for these two circumstances by checking if `expression` contains `/`, but not able to figure out a promising idea for the second circumstance since we still need that `directory` argument.

Reply:
But those are not the only two possibilities -- surely some of the paths could be relative and some could be absolute. And this check doesn't tell you anything -- if the whole expression contains a `/` it could be anywhere. It could be in the middle of a relative path, like `subdirectory/filename.fits`. 

You shouldn't be guessing whether the directory is required -- if it's not required it won't be used anyway. If the user provides a directory, you should pass in that directory. But make it optional, and set it to one of the default directories if it's `None`.

If you're going to use `pwd` (which is a good idea), make sure to change this in the hypercube function as well (it's currently using the GUI browser's starting directory, which may not be the same). The documentation for this parameter should be something like `The base directory for the LEL expression, either relative to the session's current directory or an absolute path relative to the CARTA backend's root directory. Defaults to the session's current directory.`

Reply:
Sorry; that last part is a little misleading. We have established that in the hypercube function that directory is never actually used (because the individual images have their own separate directories). So changing it should have no effect. But I would like to remove the code which sets it to the GUI browser's starting directory, because we should never be defaulting to that in the wrapper code (we should use the session's `pwd` instead), and leaving it in may be confusing later.

I have remembered that we should be normalizing *this* directory parameter (which may be used), like we do in the other functions: `directory = self.resolve_file_path(directory)`. We do this to convert all relative paths to absolute paths using `pwd` as the base (that's how we can make everything relative to our `pwd` and bypass the GUI browser's current directory, which changes when we open a file).